webodf/lib/ops/OperationTransformMatrix.js

```diff
@@ -274,6 +274,262 @@ ops.OperationTransformMatrix = function OperationTransformMatrix() {
     /* Transformation methods */
 
     /**
+     * @param {!ops.OpAddAnnotation.Spec} addAnnotationSpecA
+     * @param {!ops.OpAddAnnotation.Spec} addAnnotationSpecB
+     * @param {!boolean} hasAPriority
+     * @return {?{opSpecsA:!Array.<!Object>, opSpecsB:!Array.<!Object>}}
+     */
+    function transformAddAnnotationAddAnnotation(addAnnotationSpecA, addAnnotationSpecB, hasAPriority) {
+        var aBeforeB;
+
+        if (addAnnotationSpecA.position < addAnnotationSpecB.position) {
+            aBeforeB = true;
+        } else if (addAnnotationSpecB.position < addAnnotationSpecA.position) {
+            aBeforeB = false;
+        } else {
+            aBeforeB = hasAPriority;
+        }
+
+        if (aBeforeB) {
```

Comment:
If you wanted to de-dup this logic a bit, you could break it into ``firstAnnotation`` + ``secondAnnotation`` vars, and apply the same logic based on ``aBeforeB``. It's all by ref, so nothing stopping you keeping the return block the same as it is now, just prevents the duplicate lines.